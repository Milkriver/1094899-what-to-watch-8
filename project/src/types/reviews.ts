export interface IReview {
    id: number,
    movieId: number,
    author: string,
    date: string,
    rating: number,
    text: string,
  }
