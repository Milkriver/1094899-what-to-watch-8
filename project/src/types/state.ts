import { AuthorizationStatus } from '../const';
import { IMovie } from './common';
import { IReviewResponse } from './reviews';

export interface IState {
  currentGenre: string
  movies: IMovie[]
  isDataLoaded: boolean
  authorizationStatus: AuthorizationStatus,
  movie: IMovie,
  sameMovies: IMovie[],
  reviews: IReviewResponse[],
  promoMovie: IMovie,
}
