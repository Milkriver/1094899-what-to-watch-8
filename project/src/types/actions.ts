import { AxiosInstance } from 'axios';
import { Action } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { IState } from './state';

export enum ActionType {
  ChangeGenre = 'movieList/ChangeGenre',
  LoadMovies = 'movieList/LoadMovies',
  LoadSingleMovie = 'movieCard/LoadSingleMovie',
  LoadActiveMovie = 'movieCard/LoadActiveMovie',
  LoadSameGenreMovies = 'movieCard/LoadSameGenreMovies',
  LoadPromoMovie = 'movieCard/LoadPromoMovie',
  LoadFavouriteMovies = 'userList/LoadFavouriteMovies',
  AddFavoriteMovie = 'userList/AddFavoriteMovie',
  RemoveFavoriteMovie = 'userList/RemoveFavoriteMovie',
  LoadReviews = 'movieCard/LoadReviews',
  AddReview = 'movieCard/AddReview',
  RequireAuthorization = 'user/requireAuthorization',
  RequireLogout = 'user/requireLogout',
  LoadUserData = 'user/loadUserData',
}

export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, IState, AxiosInstance, Action>;

export type ThunkAppDispatch = ThunkDispatch<IState, AxiosInstance, Action>;
