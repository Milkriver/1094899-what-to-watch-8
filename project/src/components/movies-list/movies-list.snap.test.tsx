import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { IMovie } from '../../types/common';
import MoviesList from './movies-list';
import { configureMockStore } from '@jedmao/redux-mock-store';

const MOCK__MOVIE_1: IMovie = { id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '', backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''], runTime: 1, genre: '', released: 1, isFavorite: true };
const MOCK__MOVIE_2: IMovie = { id: 2, name: '', posterImage: '', previewImage: '', backgroundImage: '', backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 2, scoresCount: 2, director: '', starring: ['', ''], runTime: 1, genre: '', released: 2, isFavorite: false };
const MOCK__MOVIES_ARRAY: IMovie[] = [MOCK__MOVIE_2, MOCK__MOVIE_1];
const mockFn = () => Promise.resolve();
describe('Component: MoviesList', () => {

  const mockStore = configureMockStore();
  const store = mockStore({
    MOVIES: {
      movie: {
        id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '',
        backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''],
        runTime: 1, genre: '', released: 1, isFavorite: true,
      },
    },
    USER: {},
    REVIEWS: {},
  });

  store.dispatch=jest.fn();

  test('should render correctly', () => {
    const history = createMemoryHistory();
    const page = (
      <Provider store={store}>
        <Router history={history}>
          <MoviesList
            cards={MOCK__MOVIES_ARRAY}
            changeLimit={mockFn}
            moviesShowingLimit={2}
          />;
        </Router>
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
