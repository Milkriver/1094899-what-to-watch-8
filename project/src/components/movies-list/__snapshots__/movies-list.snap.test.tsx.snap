// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MoviesList should render correctly 1`] = `<div />`;
